### PR TITLE
[Connectors API] Add docs for set connector sync job stats API endpoint

### PR DESCRIPTION
**Pull request by timgrein**
Add docs for the `PUT _connector/_sync_job/my-connector-sync-job/_stats` API endpoint.

### CODE REVIEW

File: docs/reference/connector/apis/set-connector-sync-job-stats-api.asciidoc [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+* To sync data using connectors, it's essential to have the Elastic connectors service running.
+* The `connector_sync_job_id` parameter should reference an existing connector sync job.
+
+[[set-connector-sync-job-stats-api-desc]]
+==== {api-description-title}
+
+Sets the stats for a connector sync job.
+Stats include: `deleted_document_count`, `indexed_document_count`, `indexed_document_volume` and `total_document_count`.
+`last_seen` can also be update using this API.
```
Comment (leemthompo) [inline docs/reference/connector/apis/set-connector-sync-job-stats-api.asciidoc:24]:
```suggestion
`last_seen` can also be updated using this API.
```

File: docs/reference/connector/apis/set-connector-sync-job-stats-api.asciidoc [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+`<connector_sync_job_id>`::
+(Required, string)
+
+[role="child_attributes"]
+[[set-connector-sync-job-stats-api-request-body]]
+==== {api-request-body-title}
+
+`deleted_document_count`::
+(Required, int) The document count the sync job deleted.
```
Comment (leemthompo) [inline docs/reference/connector/apis/set-connector-sync-job-stats-api.asciidoc:38]:
```suggestion
(Required, int) The number of documents the sync job deleted.
```

File: docs/reference/connector/apis/set-connector-sync-job-stats-api.asciidoc [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+`indexed_document_count`::
+(Required, int) The document count the sync job indexed.
```
Comment (leemthompo) [inline docs/reference/connector/apis/set-connector-sync-job-stats-api.asciidoc:41]:
```suggestion
(Required, int) The number of documents the sync job indexed.
```

File: docs/reference/connector/apis/set-connector-sync-job-stats-api.asciidoc [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+`indexed_document_volume`::
+(Required, int) The document volume the sync job indexed.
```
Comment (leemthompo) [inline docs/reference/connector/apis/set-connector-sync-job-stats-api.asciidoc:44]:
```suggestion
(Required, int) The total size of the data the sync job indexed.
```

consider adding units

File: docs/reference/connector/apis/set-connector-sync-job-stats-api.asciidoc [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+`total_document_count`::
+(Optional, int) The total document count in the target index after the sync job finished.
```
Comment (leemthompo) [inline docs/reference/connector/apis/set-connector-sync-job-stats-api.asciidoc:47]:
```suggestion
(Optional, int) The total number of documents in the target index after the sync job finished.
```

File: docs/reference/connector/apis/set-connector-sync-job-stats-api.asciidoc [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+
+`last_seen`::
+(Optional, instant) The timestamp to set `last_seen` of the connector sync job to.
```
Comment (leemthompo) [inline docs/reference/connector/apis/set-connector-sync-job-stats-api.asciidoc:50]:
```suggestion
(Optional, instant) The timestamp to set the connector sync job's `last_seen` property.
```

property or attribute?

File: docs/reference/connector/apis/set-connector-sync-job-stats-api.asciidoc [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+
+[[set-connector-sync-job-stats-api-response-codes]]
+==== {api-response-codes-title}
```
Comment (leemthompo) [inline docs/reference/connector/apis/set-connector-sync-job-stats-api.asciidoc:53]: Guess we should have a 404 too right?

**Review by leemthompo — COMMENTED** [review]
Few small rewordings and question

File: docs/reference/connector/apis/set-connector-sync-job-stats-api.asciidoc (hunk 6 above)
Comment (timgrein) [reply docs/reference/connector/apis/set-connector-sync-job-stats-api.asciidoc]: Can you explain the difference between the terms in the context of docs? I'm not familiar with it 🤔

File: docs/reference/connector/apis/set-connector-sync-job-stats-api.asciidoc (hunk 7 above)
Comment (timgrein) [reply docs/reference/connector/apis/set-connector-sync-job-stats-api.asciidoc:53]: Yes good catch, added it with https://github.com/elastic/elasticsearch/pull/103469/commits/9a489f92a1a977c40d1a24148ee0f7624de4d30a

File: docs/reference/connector/apis/set-connector-sync-job-stats-api.asciidoc (hunk 4 above)
Comment (timgrein) [reply docs/reference/connector/apis/set-connector-sync-job-stats-api.asciidoc]: Good point, added with https://github.com/elastic/elasticsearch/pull/103469/commits/68903fdd6eaa50a207f167aacbb073f19cb8c100

**Review by leemthompo — APPROVED** [review]
LGTM 🚀

File: docs/reference/connector/apis/set-connector-sync-job-stats-api.asciidoc (hunk 6 above)
Comment (leemthompo) [reply docs/reference/connector/apis/set-connector-sync-job-stats-api.asciidoc]: not sure either what the technical term for what `last_seen` is in this context, i.e. I think it's a "property" of the sync job...

**Pull request merged**